src/fmu/dataio/_readers/tsurf_file_reader.py

```diff
@@ -0,0 +1,340 @@
+        return v
+
+
+@dataclass
+class TSurfFileData:
+    """Data class for TSurf file data.
+    TSurf is a file format for triangulations used in for example
+    the GOCAD software. RMS can export the surfaces in the structural model
+    in the TSurf format.
+    """
+
+    # TODO: export as simple export in fmu-dataio. See 'readers.py: faultroom'
+    # and 'volumetrics' for examples.
+    # Will thus have one class for reading/writing files and storing file data,
+    # and another class for SUMO export.
+    # The reader/writer class can thus be used independently of fmu-dataio.Export()
+
+    # All fields are required, no more fields are allowed in this file format
+
+    # TODO: currently you can set 'header' and 'coordinate_system' to values
+    # of any type. This would result in an invalid TSurf file.
+    # The TSurfFileData class is not a Pydantic BaseModel because using np.ndarray for
+    # vertices and triangles is not natively supported by Pydantic. It is possible
+    # to use a custom validator, but it is more complicated.
+    # To validate 'header' and 'coordinate system', create __post_init__().
+    # If wrong types, raise an error.
+    header: Header
+    coordinate_system: CoordinateSystem
+    vertices: np.ndarray
+    triangles: np.ndarray
+
+    # https://datascience.statnett.no/2020/05/11/how-we-validate-data-using-pydantic/
+    # TODO: verify that using 'class Config' has the effect that other fields are
+    # not allowed (catch typos, and makes it excplicit that if new features are needed
+    # they must be handled here)
+    class Config:
+        extra = "forbid"
+
+
+def extract_header(lines: List[str]) -> Header:
+    for line in lines:
+        match = re.match(r"^(\w+):\s*(.*)$", line)
+        if match and match.group(1).lower() == "name":
+            name = match.group(2)
+        else:
+            raise ValueError("Header section is missing the 'name' field.")
+    return Header(name=name)
+
+
+def extract_coordinate_system(lines: List[str]) -> CoordinateSystem:
+    # TODO: remove debug_print
+    debug_print = False
+
+    cs = {}
+    for line in lines:
+        match = re.match(r'^(\w+)\s+"(.+)"\s+"(.+)"\s+"(.+)"$', line)
+        if debug_print:
+            print("********* match 1: ", match)
+        if match:
+            # coord_sys[match.group(1).lower()] = match.groups()[1:]
+            cs[match.group(1)] = match.groups()[1:]
+        else:
+            match = re.match(r"^(\w+)\s+(.*)$", line)
+            if debug_print:
+                print("********* match 2: ", match)
+            if match:
+                # coord_sys[match.group(1).lower()] = match.group(2)
+                cs[match.group(1)] = match.group(2)
+
+    return CoordinateSystem(
+        name=cs["NAME"],
+        axis_name=cs["AXIS_NAME"],
+        axis_unit=cs["AXIS_UNIT"],
+        zpositive=cs["ZPOSITIVE"],
+    )
+
+
+def parse_vertices_section(lines: List[str]) -> np.ndarray:
```

Comment:
Make private `_ parse_vertices_section(...)` (this is also the case for similar function only used internally in this module)